lib/samson/build_finder.rb

```diff
@@ -28,7 +28,10 @@ def cancelled!
     def ensure_successful_builds
       builds = find_or_create_builds
       builds.compact.each do |build|
-        wait_for_build_completion(build)
+        payload = { project: build.project.name, gcb: build.external? }
```

Comment:
external does not necessary mean GCB since builds could come from anywhere ... so maybe use `external:` ?